src/net/ip4/udp.cpp

```diff
@@ -34,22 +35,49 @@ namespace net {
 
   void UDP::receive(net::Packet_ptr pckt)
   {
-    auto udp = static_unique_ptr_cast<PacketUDP>(std::move(pckt));
+    auto udp_packet = static_unique_ptr_cast<PacketUDP>(std::move(pckt));
 
     debug("<%s> UDP", stack_.ifname().c_str());
     debug("\t Source port: %u, Dest. Port: %u Length: %u\n",
-          udp->src_port(), udp->dst_port(), udp->length());
+          udp_packet->src_port(), udp_packet->dst_port(), udp_packet->length());
 
-    auto it = ports_.find(udp->dst_port());
+    auto it = ports_.find(udp_packet->dst_port());
     if (LIKELY(it != ports_.end())) {
       debug("<%s> UDP found listener on port %u\n",
-              stack_.ifname().c_str(), udp->dst_port());
-      it->second.internal_read(std::move(udp));
+              stack_.ifname().c_str(), udp_packet->dst_port());
+      it->second.internal_read(std::move(udp_packet));
       return;
     }
 
     debug("<%s> UDP: nobody listening on %u. Drop!\n",
-            stack_.ifname().c_str(), udp->dst_port());
+            stack_.ifname().c_str(), udp_packet->dst_port());
+
+    // Sending ICMP error message of type Destination Unreachable and code PORT
+    // But only if the destination IP address is not broadcast or multicast
+    auto ip4_packet = static_unique_ptr_cast<PacketIP4>(std::move(udp_packet));
+    if (ip4_packet->ip_dst() != IP4::ADDR_BCAST and (ip4_packet->ip_dst().part(3) <= 224 or
+      ip4_packet->ip_dst().part(3) >= 239))
+    {
+      stack_.icmp().destination_unreachable(std::move(ip4_packet), icmp4::code::Dest_unreachable::PORT);
+    }
+  }
+
+  void UDP::error_report(Error_type type, Error_code code,
```

Comment:
Signature will change to e.g. `UDP::error_report(ICMP_Error)` or something, as discussed